### PR TITLE
CCIP-3555  Providing evm specific implementation of tokenDataEncoder

### PR DESCRIPTION
**Pull request by mateusz-sekara**
### Requires

Please see https://github.com/smartcontractkit/chainlink-ccip/pull/186 for details

### CODE REVIEW

File: core/capabilities/ccip/ccipevm/tokendata.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+type EVMTokenDataEncoder struct{}
+
+func NewEVMTokenDataEncoder() EVMTokenDataEncoder {
+	return EVMTokenDataEncoder{}
+}
+
+func (e EVMTokenDataEncoder) EncodeUSDC(_ context.Context, message cciptypes.Bytes, attestation cciptypes.Bytes) (cciptypes.Bytes, error) {
+	return abihelpers.EncodeAbiStruct[usdcAttestationPayload](usdcAttestationPayload{
```
Comment (makramkd) [inline core/capabilities/ccip/ccipevm/tokendata.go]: The type argument is unnecessary I think

File: core/capabilities/ccip/ccipevm/tokendata_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		{
+			name:        "empty attestation",
+			message:     []byte("message"),
+			attestation: nil,
+		},
+		{
+			message:     []byte("message"),
+			attestation: []byte("attestation"),
+		},
+	}
```
Comment (makramkd) [inline core/capabilities/ccip/ccipevm/tokendata_test.go:46]: Small nit: you have real-world examples of messages and attestations right? Can we put some of them here, wdyt?

Comment (mateusz-sekara) [reply core/capabilities/ccip/ccipevm/tokendata_test.go:46]: Let me try looking for some examples

Comment (makramkd) [reply core/capabilities/ccip/ccipevm/tokendata_test.go:46]: You pulled some real-world examples in for the tests in chainlink-ccip, but maybe I'm mistaking those for something else.

**Review by makramkd — DISMISSED** [review]

**Review by makramkd — APPROVED** [review]

**Review by dimkouv — APPROVED** [review]

**Pull request merged**